acvm/src/pwg.rs

```diff
@@ -30,37 +42,31 @@ pub fn witness_to_value(
         None => Err(OpcodeNotSolvable::MissingAssignment(witness.0).into()),
     }
 }
+
+pub fn expression_to_const(expr: &Expression) -> Option<FieldElement> {
+    expr.is_const().then_some(expr.q_c)
+}
+
 // TODO: There is an issue open to decide on whether we need to get values from Expressions
 // TODO versus just getting values from Witness
 pub fn get_value(
     expr: &Expression,
     initial_witness: &BTreeMap<Witness, FieldElement>,
-) -> Result<FieldElement, OpcodeResolutionError> {
-    let mut result = expr.q_c;
-
-    for term in &expr.linear_combinations {
-        let coefficient = term.0;
-        let variable = term.1;
-
-        // Get the value assigned to that variable
-        let assignment = *witness_to_value(initial_witness, variable)?;
-
-        result += coefficient * assignment;
-    }
-
-    for term in &expr.mul_terms {
-        let coefficient = term.0;
-        let lhs_variable = term.1;
-        let rhs_variable = term.2;
-
-        // Get the values assigned to those variables
-        let lhs_assignment = *witness_to_value(initial_witness, lhs_variable)?;
-        let rhs_assignment = *witness_to_value(initial_witness, rhs_variable)?;
+) -> Option<FieldElement> {
+    expression_to_const(&ArithmeticSolver::evaluate(expr, initial_witness))
+}
 
-        result += coefficient * lhs_assignment * rhs_assignment;
+pub fn get_value_unwrap(
+    expr: &Expression,
+    initial_witness: &BTreeMap<Witness, FieldElement>,
+) -> Result<FieldElement, OpcodeResolutionError> {
+    let expr = ArithmeticSolver::evaluate(expr, initial_witness);
```

Comment:
same comment about `_unwrap`, I'm not sure why we need these instead of just using the original get_value that returned a Result

Reply:
I guess it made more sens with the GateResolution change, I don't remember.
I have restored the previous name and removed the other.